app/src/test/java/com/jalasoft/routesapp/ui/auth/registerUser/viewModel/RegisterUserViewModelTest.kt

```diff
@@ -1,38 +1,279 @@
+
+        try {
+            viewModel.errorMessage.observeForever(observer)
+            viewModel.registerUserAuth(names, email, password, false)
+
+            val value = viewModel.errorMessage.getOrAwaitValue()
+            assertNotNull(value)
+        } finally {
+            viewModel.errorMessage.removeObserver(observer)
+        }
+    }
+
+    @Test()
+    fun `Given a valid User and is registered in UserAuth it register the User in the database`() {
+        runBlocking {
+            val observer = Observer<Boolean> {}
+
+            try {
+                viewModel.registerUser.observeForever(observer)
+                viewModel.registerUser(names, email, UserTypeLogin.NORMAL)
+                val result = fakeManager.createUser(names, email, typeLogin)
+
+                assertEquals(result.data.toString(), names)
+
+                val value = viewModel.registerUser.getOrAwaitValue()
+                assertThat(value, `is`(true))
+            } finally {
+                viewModel.registerUser.removeObserver(observer)
+            }
+        }
+    }
+
+    @Test()
+    fun `Given Google Email is verify in case the Email is not register it register the User and SignIn`() {
+        runBlocking {
+            viewModel.userGoogleAuth(names, email, typeLogin, credential, true)
+            val result = fakeManager.createUser(names, email, typeLogin)
+            assertEquals(result.data.toString(), names)
+
+            val observer = Observer<Boolean> {}
+            try {
+                viewModel.signInGoogle.observeForever(observer)
+                val value = viewModel.signInGoogle.getOrAwaitValue()
+                assertThat(value, `is`(true))
+            } finally {
+                viewModel.signInGoogle.removeObserver(observer)
+            }
+        }
+    }
+
+    @Test()
+    fun `Given Google Email is verify in case the Email is registered it SignIn`() {
+        runBlocking {
+            viewModel.userGoogleAuth(names, email, typeLogin, credential, false)
+
+            val observer = Observer<Boolean> {}
+            try {
+                viewModel.signInGoogle.observeForever(observer)
+                val value = viewModel.signInGoogle.getOrAwaitValue()
+                assertThat(value, `is`(true))
+            } finally {
+                viewModel.signInGoogle.removeObserver(observer)
+            }
+        }
+    }
+
+    @Test()
+    fun `Given Google Email is not register it register the User and SignIn`() {
+        runBlocking {
+            viewModel.registerUserWithGoogle(names, email, typeLogin, credential)
+            val result = fakeManager.createUser(names, email, typeLogin)
+            assertEquals(result.data.toString(), names)
+
+            val observer = Observer<Boolean> {}
+            try {
+                viewModel.signInGoogle.observeForever(observer)
+                val value = viewModel.signInGoogle.getOrAwaitValue()
+                assertThat(value, `is`(true))
```

Comment:
wouldn't this work only with assertTrue?

Reply:
or assertEquals true

Reply:
i'll check it

Reply:
if it works, change it that way for all the tests that use this

Reply:
Done!